demo/full/bxf-parser2/tmp/hapi.js

```diff
@@ -0,0 +1,417 @@
+const BXFUrl = "http://127.0.0.1:8080/playlist_jeudi.xml";
+const BXFParser = "http://127.0.0.1:3000/playlist";
+const blackManifestURL = "http://127.0.0.1:8081/black.mpd";
+const imageDictionnary = {
+  "Déconseillé -10ans": "http://127.0.0.1:8082/picto10.png",
+}
+
+function getURLForLogoTitle(title) {
+  return imageDictionnary[title];
+}
+
+function getLicenseURL(initData) {
+  const keyid = parseInitData(initData).toUpperCase();
+  if (!keyid) {
+    throw "ERROR";
+  }
+  return kidByAffairePgrm[keyid];
+}
+
+function getLicense(challenge, _, initData) {
+  const licenseURL = getLicenseURL(initData);
+  return new Promise((resolve) => {
+    getLicenseObject(token, licenseURL, challenge).then((license) => {
+      resolve(license);
+    })
+  });
+}
+
+function getServerCertifcate() {
+  const xhr = new XMLHttpRequest();
+  xhr.open("GET", "https://secure-webtv-static.canal-plus.com/widevine/cert/cert_license_widevine_com.bin", true);
+  xhr.responseType = "arraybuffer";
+  return new Promise((resolve) => {
+    xhr.onload = (evt) => {
+      resolve(evt.target.response);
+    }
+    xhr.send();
+  })
+}
+
+function parseInitData(initData) {
+  const textDecoder = new TextDecoder("utf-8");
+  function getAllIndexes(arr, val) {
+    var indexes = [], i = -1;
+    while ((i = arr.indexOf(val, i+1)) != -1){
+        indexes.push(i);
+    }
+    return indexes;
+  }
+  const idx = getAllIndexes(textDecoder.decode(initData), "pssh");
+  const psshs = idx.reduce((acc, value) => {
+    const begin = value - 4;
+    let position = begin; 
+    position += 8;
+    const version = be4toi(initData, position);
+    position += 4;
+    if (version === 0) {
+      position += 16;
+      const privateLength = be4toi(initData, position);
+      position += 4;
+      const privateData = initData.subarray(position, position + privateLength);
+      acc.push(privateData);
+    }
+    return acc;
+  }, []);
+  const widevinePssh = psshs.find((pssh) => pssh.length === 20);
+  if (widevinePssh) {
+    return bytesToHex(widevinePssh.subarray(4, 20), "-");
+  }
+  return undefined;
+}
+
+parseBXF(BXFUrl, BXFParser, "text").then(async ({ contents: parsedContents }) => {
+  console.log("############## Parsed BXF ################");
+  const contents = [];
+  const overlays = [];
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Either use this collection's contents or remove the collection. ([javascript:S4030](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS4030))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z3fW3Y69bz5aGW6&open=AWbt7Z3fW3Y69bz5aGW6)

[ISSUE_KEY]: AWbt7Z3fW3Y69bz5aGW6